### PR TITLE
KAFKA-3589: set inner serializer for ChangedSerde upon initialization

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
Also refactored a few mockers for the unit tests. cc @ewencp @ijuma @enothereska

File: streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java [hunk 1]
```diff
@@ -115,6 +115,7 @@ public ProcessorNode build(String applicationId) {
         private Deserializer keyDeserializer;
         private Deserializer valDeserializer;
 
+
```
Comment (enothereska) [inline streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java]: nit: do we need this white line?

**Comment by enothereska** [comment]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedDeserializer.java [hunk 2]
```diff
@@ -26,12 +26,20 @@
...
     private static final int NEWFLAG_SIZE = 1;
 
-    private final Deserializer<T> inner;
+    private Deserializer<T> inner;
 
     public ChangedDeserializer(Deserializer<T> inner) {
         this.inner = inner;
     }
 
+    public Deserializer<T> getInner() {
```
Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedDeserializer.java]: Shouldn't this just be `inner()`? We normally don't use `get*` in Kafka.

Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedDeserializer.java]: Ack.

File: streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java (hunk 1 above)
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java]: Ack.

**Comment by guozhangwang** [comment]
Thanks for the reviews. Merged to trunk.

**Pull request closed without merging**